File: progra3-0583-equipo5-frontend/Frontend_C#/SoftPac/SoftPacPersistance/DAO/UsuariosDAO.cs
using System;
using System.Collections.Generic;
using System.Linq;
using System.Text;
using System.Threading.Tasks;
using SoftPac.Model;

namespace SoftPac.Persistance.DAO
{
    public interface UsuariosDAO
    {
        int Insertar(UsuariosDTO usuario);
        int Modificar(UsuariosDTO usuario);
        int eliminar(UsuariosDTO usuario);
        int eliminarLogico(UsuariosDTO usuario);
        UsuariosDTO ObtenerPorId(int usuarioId);
        IList<UsuariosDTO> ListarTodos();
        UsuariosDTO ObtenerPorCorreo(String correo);

        UsuariosDTO ObtenerPorNombreUsuario(string nombreUsuario); // <-- AÑADIR ESTA LÍNEA

    }
}

File: Backend_Java/SoftPacPersistance/src/main/java/pe/edu/pucp/softpac/dao/UsuariosDAO.java
package pe.edu.pucp.softpac.dao;

import pe.edu.pucp.softpac.model.UsuariosDTO;
import java.util.List;

public interface UsuariosDAO {
    public Integer insertar(UsuariosDTO usuario);

    public UsuariosDTO obtenerPorId(Integer usuarioId);

    public List<UsuariosDTO> listarTodos();

    public Integer modificar(UsuariosDTO usuario);

    public Integer eliminar(UsuariosDTO usuario);

    public UsuariosDTO obtenerPorCorreo(String correo);

    public Integer eliminarLogico(UsuariosDTO usuario);
    
    public UsuariosDTO obtenerPorNombreDeUsuario(String nombreDeUsuario);
    
}
